### PR TITLE
[arduino] Fixed library compilation issues & changed file extensions.

### PR DESCRIPTION
**Pull request by therealhuy**
Changed all file extensions to work with Arduino. Included Arduino.h in cpp files.

### CODE REVIEW

File: wallu/core/motor/motor.cpp [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #include "motor.hh"
```
Comment (abdullahelqaq) [inline wallu/core/motor/motor.cpp]: You renamed this to `motor.h`

File: wallu/core/motor/motor.cpp [hunk 2]
```diff
@@ -23,16 +24,19 @@ void Motor::set_throttle(int pwm_value, MotorDirection dir)
         m_pwm = 0;
         digitalWrite(m_pins.backward, LOW);
         digitalWrite(m_pins.forward, LOW);
+        break;
```
Comment (abdullahelqaq) [inline wallu/core/motor/motor.cpp:27]: Lmao that's funny, my bad for missing these

**Review by abdullahelqaq — CHANGES_REQUESTED** [review]
Small change. Thanks for catching these, I guess I'll revert to the boring file extensions :(

File: wallu/core/motor/motor.cpp (hunk 1 above)
Comment (benjaminruiz99) [inline wallu/core/motor/motor.cpp]: Should be motor.h

**Review by benjaminruiz99 — CHANGES_REQUESTED** [review]

**Review by benjaminruiz99 — APPROVED** [review]

**Review by abdullahelqaq — APPROVED** [review]
Remember to squash and merge, instead of creating a new merge commit 💯

**Pull request merged**